app/scripts/controllers/sign.ts

```diff
@@ -63,9 +64,10 @@ export type CoreMessage = AbstractMessage & {
   messageParams: AbstractMessageParams;
 };
 
-export type StateMessage = Required<AbstractMessage> & {
+export type StateMessage = Required<
+  Omit<AbstractMessage, 'securityProviderResponse'>
```

Comment:
I'm still not sure what ew gain by omitting `securityProviderResponse`?
Is it just linting errors in the migrate logic, in which case we could also just cast to avoid?